packages/components/src/DateTimePickers/DateTimePicker/DateTimePicker.test.js

```diff
@@ -11,32 +11,273 @@ describe('DateTimePicker', () => {
+
+			expect(wrapper.find(DateTimeView).exists()).toBe(true);
+			expect(wrapper.find(MonthYearView).exists()).toBe(false);
+		});
+
+		it('should render with the MonthYearView on state', () => {
+			const wrapper = shallow(<DateTimePicker
+				onSubmit={() => {}}
+			/>);
+
+			wrapper.setState({
+				isDateTimeView: false,
+			});
+
+			expect(wrapper.find(DateTimeView).exists()).toBe(false);
+			expect(wrapper.find(MonthYearView).exists()).toBe(true);
+		});
+
+		it('should switch state to MonthYearView when header title of DateTimeView is clicked', () => {
+			const wrapper = shallow(<DateTimePicker
+				onSubmit={() => {}}
+			/>);
+
+			wrapper.setState({
+				isDateTimeView: true,
+			});
+
+			const clickTitleHandler = wrapper.find(DateTimeView).prop('onClickTitle');
+			clickTitleHandler();
+			expect(wrapper.state('isDateTimeView')).toBe(false);
+		});
+
+		it('should switch state to DateTimeView when header back action of MonthYearView is clicked', () => {
+			const wrapper = shallow(<DateTimePicker
+				onSubmit={() => {}}
+			/>);
+
+			wrapper.setState({
+				isDateTimeView: false,
+			});
+
+			const clickBackHandler = wrapper.find(MonthYearView).prop('onClickBack');
+			clickBackHandler();
+			expect(wrapper.state('isDateTimeView')).toBe(true);
+		});
+	});
+
+	it('should have the last "date" data between props and selection in state', () => {
+		const d1 = new Date(2018, 2, 5);
+		const d2 = new Date(2019, 11, 21);
+		const d3 = new Date(2015, 6, 27);
+		const wrapper = shallow(<DateTimePicker
+			selectedDate={d1}
+			onSubmit={() => {}}
+		/>);
 
 		wrapper.setState({
 			isDateTimeView: true,
 		});
 
-		expect(wrapper.find(DateTimeView).exists()).toBe(true);
-		expect(wrapper.find(MonthYearView).exists()).toBe(false);
+		const dateTimeView = wrapper.find(DateTimeView);
+		expect(wrapper.state('selectedDate')).toBe(d1);
+
+		dateTimeView.prop('onSelectDate')(d2);
+		expect(wrapper.state('selectedDate')).toBe(d2);
+
+		wrapper.setProps({
+			selectedDate: d3,
+		});
+		expect(wrapper.state('selectedDate')).toBe(d3);
 	});
 
-	it('should render with the MonthYearView', () => {
-		const wrapper = shallow(<DateTimePicker />);
+	it('should have the last "time" data between props and selection in state', () => {
+		const t1 = 810;
+		const t2 = 950;
+		const t3 = 1270;
+		const wrapper = shallow(<DateTimePicker
+			selectedTime={t1}
+			onSubmit={() => {}}
+		/>);
 
 		wrapper.setState({
-			isDateTimeView: false,
+			isDateTimeView: true,
 		});
 
-		expect(wrapper.find(DateTimeView).exists()).toBe(false);
-		expect(wrapper.find(MonthYearView).exists()).toBe(true);
+		const dateTimeView = wrapper.find(DateTimeView);
+		expect(wrapper.state('selectedTime')).toBe(t1);
+
+		dateTimeView.prop('onSelectTime')(t2);
+		expect(wrapper.state('selectedTime')).toBe(t2);
+
+		wrapper.setProps({
+			selectedTime: t3,
+		});
+		expect(wrapper.state('selectedTime')).toBe(t3);
+	});
+
+	// TODO: Enable when the submit button will be implemented
```

Comment:
??

Reply:
Remove it, this TODO is not useful, when you want to implement the button, you'll just do what is needed

Reply:
✔